pkg/handlers/decide_test.go

```diff
@@ -0,0 +1,311 @@
+	suite.mux.ServeHTTP(rec, req)
+
+	suite.Equal(http.StatusOK, rec.Code)
+
+	// Unmarshal response
+	var actual client.OptimizelyDecision
+	err := json.Unmarshal(rec.Body.Bytes(), &actual)
+	suite.NoError(err)
+
+	expected := client.OptimizelyDecision{
+		UserContext:  client.OptimizelyUserContext{UserID: "testUser", Attributes: map[string]interface{}{}},
+		FlagKey:      "one",
+		RuleKey:      "1",
+		Enabled:      true,
+		VariationKey: "2",
+		Reasons:      []string{},
+	}
+	suite.Equal(expected, actual)
+
+	events := suite.tc.GetProcessedEvents()
+	suite.Equal(1, len(events))
+
+	impression := events[0]
+	suite.Equal("campaign_activated", impression.Impression.Key)
+	suite.Equal("testUser", impression.VisitorID)
+}
+
+func (suite *DecideTestSuite) TestGetFeatureMissingFeature() {
+	req := httptest.NewRequest("POST", "/decide?key=feature-missing", bytes.NewBuffer(suite.body))
+	rec := httptest.NewRecorder()
+	suite.mux.ServeHTTP(rec, req)
+	suite.Equal(http.StatusOK, rec.Code)
+
+	// Unmarshal response
+	var actual client.OptimizelyDecision
+	err := json.Unmarshal(rec.Body.Bytes(), &actual)
+	suite.NoError(err)
+
+	expected := client.OptimizelyDecision{
+		UserContext:  client.OptimizelyUserContext{UserID: "testUser", Attributes: map[string]interface{}{}},
+		FlagKey:      "feature-missing",
+		RuleKey:      "",
+		Enabled:      false,
+		VariationKey: "",
+		Reasons:      []string{"No flag was found for key \"[feature-missing]\"."},
+	}
+
+	suite.Equal(0, len(suite.tc.GetProcessedEvents()))
+	suite.Equal(expected, actual)
+}
+
+func (suite *DecideTestSuite) TestActivateFeatures() {
+	// 100% enabled rollout
+	feature := entities.Feature{Key: "featureA"}
+	suite.tc.AddFeatureRollout(feature)
+
+	// 100% enabled feature test
+	featureB := entities.Feature{Key: "featureB"}
+	suite.tc.AddFeatureTest(featureB)
+
+	// Feature test 100% enabled variation 100% with variation variable value
+	variable := entities.Variable{DefaultValue: "default", ID: "123", Key: "strvar", Type: "string"}
+	featureC := entities.Feature{Key: "featureC", VariableMap: map[string]entities.Variable{"strvar": variable}}
+	suite.tc.AddFeatureTestWithCustomVariableValue(featureC, variable, "abc_notdef")
+
+	expected := []client.OptimizelyDecision{
+		{
+			UserContext:  client.OptimizelyUserContext{UserID: "testUser", Attributes: map[string]interface{}{}},
+			FlagKey:      "featureA",
+			RuleKey:      "1",
+			Enabled:      true,
+			VariationKey: "3",
+			Reasons:      []string{},
+		},
+		{
+			UserContext:  client.OptimizelyUserContext{UserID: "testUser", Attributes: map[string]interface{}{}},
+			FlagKey:      "featureB",
+			RuleKey:      "5",
+			Enabled:      true,
+			VariationKey: "6",
+			Reasons:      []string{},
+		},
+		{
+			UserContext:  client.OptimizelyUserContext{UserID: "testUser", Attributes: map[string]interface{}{}},
+			FlagKey:      "featureC",
+			RuleKey:      "12",
+			Enabled:      true,
+			VariationKey: "13",
+			Reasons:      []string{},
```

Comment:
Aren't we supposed to have "Variables" non-empty here?  If not, can we modify the tests to validate Variables?